velox/connectors/WriteProtocol.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/vector/ComplexVector.h"
+
+namespace facebook::velox::connector {
+
+class ConnectorInsertTableHandle;
+class ConnectorQueryCtx;
+
+enum class CommitStrategy { kNoCommit, kTaskCommit };
+
+/**
+ * Interface to provide key parameters for writers. Ex., write and commit
+ * locations, append or overwrite, etc.
+ */
+class WriterParameters {
+ public:
+  virtual ~WriterParameters() = default;
+};
+
+/**
+ * Interface that includes all info from write. It will be passed to commit() of
+ * WriteProtocol.
+ */
+class WriteInfo {
+ public:
+  virtual ~WriteInfo() = default;
+};
+
+/**
+ * Abstraction for write behaviors. Systems register WriteProtocols by
+ * CommitStrategy. Writers call newWriteProtocol() to get an instance of the
+ * registered WriteProtocol when needed.
+ */
+class WriteProtocol {
+ public:
+  virtual ~WriteProtocol(){};
+
+  virtual CommitStrategy getCommitStrategy() const = 0;
+
+  virtual std::string encodeCommitStrategy() const = 0;
```

Comment:
Would you document all methods? What does encodeCommitStrategy do?

Reply:
CommitStrategy is a fixed num. Let's add free-function commitStrategyToString to convert enum value to string.